src/s3fs.cpp

```diff
@@ -287,6 +289,52 @@ static bool set_s3fs_usr2_handler()
   return true;
 }
 
+static void s3fs_usr1_handler(int sig)
+{
+  static bool isWorking = false;
+
+  if(SIGUSR1 != sig){
+    S3FS_PRN_ERR("The handler for SIGUSR1 received signal(%d)", sig);
+    return;
+  }
+  if(isWorking){
+    S3FS_PRN_ERR("Since SIGUSR1 is currently being processed, multiple processing cannot be performed.");
+    return;
+  }
+  isWorking = true;
+
+  // check all cache
+  if(!FdManager::get()->CheckAllCache()){
```

Comment:
As you say, this method is not good.
(It shouldn't even call the printf function)
This is an option only for debugging, but to avoid the possibility of leaving a problem in the future, I should consider using thread processing.
However, it is necessary to consider that s3fs is forked when implementing with thread and whether it can be safely implemented.

Reply:
Fixed the signal handler.
The check-cache processing is threaded and separated from the direct processing from the signal handler.